seed/building_sync/mappings.py

```diff
@@ -207,18 +207,106 @@ def to_energy_type(energy_type):
+        'Geothermal': 'Other:',                                     # other?
+        'Solar': 'Electric - Solar',                                # correct mapping?
+        'Biomass': 'Other:',                                        # other?
+        'Hydrothermal': 'Other:',                                   # other?
+        'Dry steam': 'Other:',                                      # other?
+        'Flash steam': 'Other:',                                    # other?
+        'Ethanol': 'Other:',                                        # other?
+        'Biodiesel': 'Other:',                                      # other?
+        'Waste heat': 'Other:',                                     # other?
+        'Dual fuel': 'Other:',                                      # other?
+        'Gasoline': 'Other:',                                       # other?
+        'Thermal-Exported': 'Other:',                               # other?
+        'Thermal-Onsite generated': 'Other:',                       # other?
+        'Other delivered-Exported': 'Other:',                       # other?
+        'Other delivered-Onsite generated': 'Other:',               # other?
+        'Other metered-Exported': 'Other:',                         # other?
+        'Other metered-Onsite generated': 'Other:',                 # other?
+        'Other': 'Other:',
+        'Unknown': 'Other:',                                        # other?
+    }.get(energy_type, energy_type).lower()
     for energy_pair in Meter.ENERGY_TYPES:
         if energy_pair[1].lower() == energy_name:
             return energy_pair[0]
 
-    return energy_type
+    # couldn't find this energy type... default to "Other:"
+    return Meter.ENERGY_TYPES.OTHER
+
+
+def to_energy_units(units):
+    """converts energy units from BuildingSync into one allowed by SEED
+
+    :param units: string, building sync units
+    :return: string
+    """
+
+    # valid energy unit values from the schema (<xs:element name="ResourceUnits">) and their maps
+    # non-trivial or non-obvious mappings currently map to "Unknown" and are flagged with a comment
+    # this mapping is important for unit generation... see "kbtu_thermal_conversion_factors"
+    return {
+        'Cubic Meters': 'cm (cubic meters)',
+        'kcf': 'kcf (thousand cubic feet)',
+        'MCF': 'Mcf (million cubic feet)',
+        'Gallons': 'Gallons (US)',              # assuming "US" for conversion_factor but could be "UK"
+        'Wh': 'Wh (Watt-hours)',
+        'kWh': 'kWh (thousand Watt-hours)',
+        'MWh': 'MWh (million Watt-hours)',
+        'Btu': 'Btu',
+        'kBtu': 'kBtu (thousand Btu)',
+        'MMBtu': 'MBtu/MMBtu (million Btu)',
+        'therms': 'therms',
+        'lbs': 'Lbs. (pounds)',
+        'Klbs': 'kLbs. (thousand pounds)',
+        'Mlbs': 'MLbs. (million pounds)',
+        'Mass ton': 'Tons',                     # assuming "Tons" over "Tonnes (metric)"
```

Comment:
Hmm, if the data is always coming from BuildingSync though, is it ok to assume it's always going to imperial/US units?

Reply:
We should add a new energy type to the schema then that covers this, say "Mass Tonnes (metric)".

Reply:
For carbon it will be kgCO2_e or tonnes CO2. 

<img width="437" alt="image" src="https://user-images.githubusercontent.com/1907354/122575120-a406e680-d00d-11eb-9272-4b107c9cc603.png">

Since we specify the units in the schema, I think we assume that the user put the data with the right units.